detection_rules/rule_validators.py

```diff
@@ -147,7 +147,7 @@ def validate_stack_combos(self, data: QueryRuleData, meta: RuleMeta) -> Union[KQ
             ecs_version = mapping['ecs']
             err_trailer = f'stack: {stack_version}, beats: {beats_version}, ecs: {ecs_version}'
 
-            beat_types, beat_schema, schema = self.get_beats_schema(data.index or [],
+            beat_types, beat_schema, schema = self.get_beats_schema(data.index_or_dataview or [],
```

Comment:
Instead of `data.index_or_dataview or []` **_does it make sense_** to just default and return`[]` in `data.index_or_dataview` if both are None? Then we wouldn't need this or condition if the several places we have it.

Reply:
We could do this, but it might imply that we support rules where neither the index or dataview are set, which I do not think that we want to support at this time.